spec/lucky/base_http_client_spec.cr

```diff
@@ -10,58 +10,120 @@ Spec.before_each do
   TestServer.reset
 end
 
+class HelloWorldAction < TestAction
+  post "/hello" do
+    plain_text "unused"
+  end
+end
+
+class MyClient < Lucky::BaseHTTPClient
+end
+
 describe Lucky::BaseHTTPClient do
-  describe "#get" do
-    it "sends requests to correct uri (with the correct query params) and gives the right response" do
-      TestServer.route("hello", "world")
-      Lucky::Server.temp_config(host: "localhost", port: 6226) do
-        client = Lucky::BaseHTTPClient.new
-        response = client.get("hello", params: HTTP::Params.new(raw_params: {"foo" => ["bar"]}))
-
-        response.body.should eq "world"
-
-        request = server.last_request.not_nil!
-        params = request.query.not_nil!
-        params.should eq HTTP::Params.encode({"foo" => "bar"})
+  describe "headers" do
+    it "sets headers and allows chaining" do
+      with_fake_server(path: "/hello", response_body: "world") do
+        MyClient.new
+          .headers(accept: "text/csv")
+          .headers(content_type: "application/json")
+          .headers("Foo": "bar")
+          .exec(HelloWorldAction)
+
+        request = server.last_request
+        request.headers["accept"].should eq("text/csv")
+        request.headers["content-type"].should eq("application/json")
+        request.headers["Foo"].should eq("bar")
       end
     end
   end
 
-  describe "#delete" do
-    it "sends requests to correct uri (with the correct query params) and gives the right response" do
-      TestServer.route("hello", "world")
-      Lucky::Server.temp_config(host: "localhost", port: 6226) do
-        client = Lucky::BaseHTTPClient.new
-        response = client.delete("hello")
+  describe "exec" do
+    describe "with Lucky::Action class" do
+      it "uses the method and path" do
+        with_fake_server(path: "/hello", response_body: "world") do
+          response = MyClient.new.exec(HelloWorldAction)
+
+          request = server.last_request
+          request.path.should eq "/hello"
+          request.method.should eq("POST")
+          request.body.not_nil!.gets_to_end.should eq(NamedTuple.new.to_json)
```

Comment:
The `request`  is an `HTTP::Request` from Crystal so I can't easily change that without monkey patching it. Since end users won't be checking the request body I don't think it is worth monkey patching it.

I do  like the idea about just using a string! That's a lot clearer